qiskit/quantum_info/states/utils.py

```diff
@@ -155,3 +154,52 @@ def _funm_svd(matrix, func):
     unitary1, singular_values, unitary2 = la.svd(matrix)
     diag_func_singular = np.diag(func(singular_values))
     return unitary1.dot(diag_func_singular).dot(unitary2)
+
+
+def partial_transpose(state, qargs):
+    """Return partially transposed density matrix.
+
+    Args:
+        state (DensityMatrix): the input state.
+        qargs (list): The subsystems to be transposed.
+
+    Returns:
+        DensityMatrix: The partially transposed density matrix.
+
+    Raises:
+        QiskitError: if input state is invalid.
+        QiskitError: if indices of subsystems are invalid
+
+    """
+    state = _format_state(state, validate=False)
+    n = len(state.dims())
+    lst = np.zeros(2**n, int)
```

Comment:
The ``DensityMatrix`` also allows qudit subsystems (see the `dims` argument 🙂), but in this implementation you assume qubits. We'd either have to add a check that the input density matrix is defined on qubits or generalize this code to work on arbitrary dimensions.

Reply:
Thank you for bringing this to our attention. We will take steps to rectify the issue.